src/client/sky.cpp

```diff
@@ -67,44 +68,52 @@ Sky::Sky(s32 id, ITextureSource *tsrc):
 	m_materials[2].MaterialType = video::EMT_TRANSPARENT_ALPHA_CHANNEL;
 	//m_materials[2].MaterialType = video::EMT_TRANSPARENT_ADD_COLOR;
 
-	m_sun_texture = tsrc->isKnownSourceImage("sun.png") ?
-		tsrc->getTextureForMesh("sun.png") : NULL;
-	m_moon_texture = tsrc->isKnownSourceImage("moon.png") ?
-		tsrc->getTextureForMesh("moon.png") : NULL;
-	m_sun_tonemap = tsrc->isKnownSourceImage("sun_tonemap.png") ?
-		tsrc->getTexture("sun_tonemap.png") : NULL;
-	m_moon_tonemap = tsrc->isKnownSourceImage("moon_tonemap.png") ?
-		tsrc->getTexture("moon_tonemap.png") : NULL;
+	// Ensures that sun and moon textures and tonemaps are correct.
+	setSkyDefaults();
+	m_sun_texture = tsrc->isKnownSourceImage(m_sun_params.texture) ?
+		tsrc->getTextureForMesh(m_sun_params.texture) : NULL;
+	m_moon_texture = tsrc->isKnownSourceImage(m_moon_params.texture) ?
+		tsrc->getTextureForMesh(m_moon_params.texture) : NULL;
+	m_sun_tonemap = tsrc->isKnownSourceImage(m_sun_params.tonemap) ?
+		tsrc->getTexture(m_sun_params.tonemap) : NULL;
+	m_moon_tonemap = tsrc->isKnownSourceImage(m_moon_params.tonemap) ?
+		tsrc->getTexture(m_moon_params.tonemap) : NULL;
 
 	if (m_sun_texture) {
 		m_materials[3] = mat;
 		m_materials[3].setTexture(0, m_sun_texture);
 		m_materials[3].MaterialType = video::EMT_TRANSPARENT_ALPHA_CHANNEL;
+		// Disables texture filtering
+		m_materials[3].setFlag(video::E_MATERIAL_FLAG::EMF_BILINEAR_FILTER, false);
+		m_materials[3].setFlag(video::E_MATERIAL_FLAG::EMF_TRILINEAR_FILTER, false);
+		m_materials[3].setFlag(video::E_MATERIAL_FLAG::EMF_ANISOTROPIC_FILTER, false);
+		// Use tonemaps if available
 		if (m_sun_tonemap)
 			m_materials[3].Lighting = true;
 	}
-
 	if (m_moon_texture) {
 		m_materials[4] = mat;
 		m_materials[4].setTexture(0, m_moon_texture);
 		m_materials[4].MaterialType = video::EMT_TRANSPARENT_ALPHA_CHANNEL;
+		// Disables texture filtering
+		m_materials[4].setFlag(video::E_MATERIAL_FLAG::EMF_BILINEAR_FILTER, false);
+		m_materials[4].setFlag(video::E_MATERIAL_FLAG::EMF_TRILINEAR_FILTER, false);
+		m_materials[4].setFlag(video::E_MATERIAL_FLAG::EMF_ANISOTROPIC_FILTER, false);
+		// Use tonemaps if available
 		if (m_moon_tonemap)
 			m_materials[4].Lighting = true;
 	}
 
-	for (v3f &star : m_stars) {
-		star = v3f(
-			myrand_range(-10000, 10000),
-			myrand_range(-10000, 10000),
-			myrand_range(-10000, 10000)
-		);
-		star.normalize();
+	for (int i = 5; i < 11; i++) {
+		m_materials[i] = mat;
+		m_materials[i].Lighting = true;
+		m_materials[i].MaterialType = video::EMT_SOLID;
 	}
-
 	m_directional_colored_fog = g_settings->getBool("directional_colored_fog");
+	m_star_params.count = 1;
```

Comment:
Hacky implementation. Please add a boolean parameter to `setStarCount` to force updating the stars if that's really needed.